src/arrayMethodSort.js

```diff
@@ -4,8 +4,41 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+    const newCompareFunc = (a, b) => {
+      let firstItem = a;
+      let secondItem = b;
+      
+      if (typeof a === 'number' && typeof b === 'number') {
+        firstItem = String(a);
+        secondItem = String(b)
+      }
+
+      if (firstItem < secondItem) {
+        return -1;
+      }
+
+      if (firstItem > secondItem) {
+        return 1;
+      }
+
+      return 0;
+    };
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
```

Comment:
The inner loop can be optimized by reducing the number of iterations as the largest elements will bubble up to the end of the array with each iteration. You can reduce the number of iterations by subtracting `i` from the length of the array in the loop condition.